File: Java/Q53.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class CharFrequency {
    public static void findCharacterFrequency(String inputString){
        if(inputString.isEmpty()){
            System.out.println("The input string is empty.");
            return;
        }
        Map<Character, Integer> freq = new HashMap<>();
        char[] characters = inputString.toCharArray();
        for(char ch : characters){
            int count = freq.getOrDefault(ch, 0);
            freq.put(ch, count + 1);
        }
        System.out.println("\nCharacter frequencies in the string '" + inputString + "':");
        for(Map.Entry<Character, Integer> entry : freq.entrySet()){
            System.out.println("'" + entry.getKey() + "' : " + entry.getValue());
        }
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a string: ");
        String userInput = scanner.nextLine();
        findCharacterFrequency(userInput);

        scanner.close();
    }
}

File: C++/Q53.cpp
// Q53.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string str;
    int freq[256] = {0};
    cout << "Enter a string: ";
    getline(cin, str);
    for (char c : str)
        freq[(unsigned char)c]++;

    cout << "Character frequencies:\n";
    for (int i = 0; i < 256; i++)
        if (freq[i] != 0)
            cout << (char)i << ": " << freq[i] << endl;
    return 0;
}
